### PR TITLE
Add environment variable for `echo` option in `create_engine`

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Add an environment variable to specify the `echo` option in `create_engine` to allow printing out SQL statements while developing a SQLAlchemy-related API.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

Ran the following command and it works as expected.

```
MLFLOW_SQLALCHEMYSTORE_ECHO=TRUE pytest tests/store/tracking/test_sqlalchemy_store.py -s
MLFLOW_SQLALCHEMYSTORE_ECHO=FALSE pytest tests/store/tracking/test_sqlalchemy_store.py -s
```

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/environment_variables.py [hunk 1]
```diff
@@ -14,6 +14,10 @@ def __init__(self, name, type_, default):
         self.type = type_
         self.default = default
 
+    @property
+    def defined(self):
```
Comment (WeichenXu123) [inline mlflow/environment_variables.py:18]: nit: using `is_defined` name looks more clear

File: mlflow/environment_variables.py [hunk 2]
```diff
@@ -34,6 +38,25 @@ def __repr__(self):
...
+    Represents a boolean environment variable.
+    """
+
+    def get(self):
+        if not self.defined:
+            return self.default
+
+        val = os.getenv(self.name)
+        lowercased = val.lower()
+        if lowercased not in ["true", "false"]:
```
Comment (WeichenXu123) [inline mlflow/environment_variables.py:52]: shall we support `0` / `1` as well ?

**Review by WeichenXu123 — APPROVED** [review]
LGTM

File: mlflow/environment_variables.py (hunk 2 above)
Comment (harupy) [reply mlflow/environment_variables.py]: sounds good, added

File: mlflow/environment_variables.py (hunk 1 above)
Comment (harupy) [reply mlflow/environment_variables.py]: sounds good, fixed

File: mlflow/environment_variables.py [hunk 3]
```diff
@@ -34,6 +38,24 @@ def __repr__(self):
...
+        if not self.is_defined:
+            return self.default
+
+        val = os.getenv(self.name)
+        lowercased = val.lower()
+        if lowercased not in ["true", "false", "1", "0"]:
+            raise ValueError(
+                f"{self.name} value must be one of ['true', 'false', '1', '0'], but got {val}"
+            )
+        return lowercased == "true"
```
Comment (WeichenXu123) [inline mlflow/environment_variables.py:56]: `return lowercased == "true" or lowercased == "1"`

File: mlflow/environment_variables.py [hunk 4]
```diff
@@ -34,6 +38,24 @@ def __repr__(self):
         return repr(self.name)
 
 
+class _BooleanEnvironmentVariable(_EnvironmentVariable):
```
Comment (WeichenXu123) [inline mlflow/environment_variables.py:41]: For this class, we can define a new `__init__` ctor, like
```
    def __init__(self, name, default):
        self.name = name
        self.type = bool
        self.default = default
        assert default in [True, False]
```

File: mlflow/environment_variables.py (hunk 3 above)
Comment (harupy) [reply mlflow/environment_variables.py:56]: thanks for the catch, fixed

File: mlflow/environment_variables.py (hunk 4 above)
Comment (harupy) [reply mlflow/environment_variables.py:41]: updated the constructor to perform validation on the default value.

**Review by WeichenXu123 — APPROVED** [review]

**Pull request merged**